src/sync/sync-api.js

```diff
@@ -60,6 +60,11 @@ export class SyncApi extends TypedEmitter {
   #hasRequestedFullStop = false
   /** @type {SyncEnabledState} */
   #previousSyncEnabledState = 'none'
+  /** @type {null | number} */
+  #previousDataHave = null
+  #autostopDataSyncAfter = Infinity
```

Comment:
nit: use of `Infinity` is a bit misleading since it's never actually used (kind of acts as an unset value) but guessing it's makes you write less type-related code

Reply:
Changed to `null | number` in 8675cd16f1af407de4eaa72bfcfb9226a8ec114b.